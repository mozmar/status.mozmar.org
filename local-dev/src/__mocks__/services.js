const services = [
    {
        display: true,
        group: 'group1',
        hasDetail: false,
        id: 1,
        link: 'test link',
        name: 'test service 1',
        order: 1,
        status: 'healthy'
    },
    {
        display: true,
        hasDetail: true,
        id: 2,
        link: 'test link',
        name: 'test service 2',
        order: 2,
        status: 'healthy'
    },
    {
        display: true,
        group: 'group1',
        hasDetail: false,
        id: 3,
        link: 'test link',
        name: 'test service 3',
        order: 3,
        status: 'warning'
    },
    {
        display: true,
        group: 'group2',
        hasDetail: false,
        id: 4,
        link: 'test link',
        name: 'test service 4',
        order: 10,
        status: 'healthy'
    }
];

export default services;
