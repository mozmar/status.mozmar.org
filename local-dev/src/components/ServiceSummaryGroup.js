import React, { Component } from 'react';

import ServiceSummary from './ServiceSummary';

class ServiceSummaryGroup extends Component {
    static propTypes = {
        name: React.PropTypes.string.isRequired,
        order: React.PropTypes.number.isRequired,
        services: React.PropTypes.array.isRequired
    }

    getServices() {
        var services = [];

        this.props.services.forEach(service => {
            if (!service.display) {
                return;
            }

            services.push(
                <ServiceSummary
                   key={service.id}
                   id={service.id}
                   status={service.status}
                   hasDetail={service.hasDetail}
                   link={service.link}
                   name={service.name}
                   order={service.order}/>
            );
        });

        services.sort((a, b) => {
            let a_order = a.props.order !== undefined ? a.props.order : 0;
            let b_order = b.props.order !== undefined ? b.props.order : 0;

            if (a_order < b_order) {
                return -1;
            } else if (a_order > b_order) {
                return 1;
            }

            return 0;
        });

        return services;
    }

    render() {
        var services = this.getServices();
        var header = null;

        if (this.props.name !== 'default') {
            header = (
                <li className="list-group-item list-group-item-info">
                    {this.props.name}
                </li>
            );
        }

        return (
            <ul className="list-group">
                {header}
                {services}
            </ul>
        );
    }
}

export default ServiceSummaryGroup;
