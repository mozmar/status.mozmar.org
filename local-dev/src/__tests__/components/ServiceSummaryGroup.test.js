import React from 'react';
import renderer from 'react-test-renderer';

import ServiceSummaryGroup from '../../components/ServiceSummaryGroup';

import services from '../../__mocks__/services';

describe('render ServiceSummaryGroup', () => {
    test('render basic ServiceSummaryGroup', () => {
        const component = renderer.create(
            <ServiceSummaryGroup name='Test Group' order={1} services={services} />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    // Test when name='default' (as header will not render)
    test('render default ServiceSummaryGroup', () => {
        const component = renderer.create(
            <ServiceSummaryGroup name='default' order={2} services={services} />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
