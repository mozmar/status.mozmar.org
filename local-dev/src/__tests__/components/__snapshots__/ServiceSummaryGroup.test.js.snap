exports[`render ServiceSummaryGroup render basic ServiceSummaryGroup 1`] = `
<ul
  className="list-group">
  <li
    className="list-group-item list-group-item-info">
    Test Group
  </li>
  <li
    className="list-group-item">
    <span
      className="status healthy">
      healthy
    </span>
    test service 1
  </li>
  <li
    className="list-group-item">
    <span
      className="status healthy">
      healthy
    </span>
    <a
      onClick={[Function]}
      style={Object {}}>
      test service 2
    </a>
  </li>
  <li
    className="list-group-item">
    <span
      className="status warning">
      warning
    </span>
    test service 3
  </li>
  <li
    className="list-group-item">
    <span
      className="status healthy">
      healthy
    </span>
    test service 4
  </li>
</ul>
`;

exports[`render ServiceSummaryGroup render default ServiceSummaryGroup 1`] = `
<ul
  className="list-group">
  <li
    className="list-group-item">
    <span
      className="status healthy">
      healthy
    </span>
    test service 1
  </li>
  <li
    className="list-group-item">
    <span
      className="status healthy">
      healthy
    </span>
    <a
      onClick={[Function]}
      style={Object {}}>
      test service 2
    </a>
  </li>
  <li
    className="list-group-item">
    <span
      className="status warning">
      warning
    </span>
    test service 3
  </li>
  <li
    className="list-group-item">
    <span
      className="status healthy">
      healthy
    </span>
    test service 4
  </li>
</ul>
`;
